pubnub-core/src/subscription/subscription.rs

```diff
@@ -55,9 +59,10 @@ impl<TRuntime: Runtime> Drop for Subscription<TRuntime> {
         // See: https://boats.gitlab.io/blog/post/poll-drop/
         self.runtime.spawn(async move {
             let drop_send_result = control_tx.send(command).await;
-            if is_drop_send_result_error(drop_send_result) {
-                panic!("Unable to unsubscribe");
-            }
+            assert!(
+                !is_drop_send_result_error(drop_send_result),
```

Comment:
I don't know why this is confusing me so badly but I think I understand now. Looks like assert will panic when the expression is true so this should be equivalent to the if statement. I'm still learning :)

Reply:
Correct.